### PR TITLE
added binance chain wallet support

### PR DESCRIPTION
**Pull request by pinpong**
https://binance-wallet.gitbook.io/binance-chain-extension-wallet/dev/get-started

### CODE REVIEW

**Comment by thanhnamitit** [comment]
Hi, I run example code on chrome. But application run on chrome debug mode, so I cannot install metamask there to connect to wallet. Could you please give me a solution to fix this?

**Comment by pinpong** [comment]
Hi,

check: https://github.com/flutter/flutter/issues/43809

**Review by treeder — APPROVED** [review]
Looks good, if you can resolve the conflict, we can merge this. Sorry for the delay, missed this in my inbox I guess.

File: lib/flutter_web3_provider.dart [hunk 1]
```diff
@@ -3,7 +3,10 @@ library flutter_web3_provider;
...
 
-Web3Provider getWeb3Provider() {
-  Web3Provider web3 = Web3Provider(ethereum);
-  return web3;
+Web3Provider? getWeb3Provider() {
+  final provider = ethereum ?? binanceChain ?? web3;
+  if (provider != null) {
+    return Web3Provider(provider);
+  }
+  return null;
```
[the remark this answers is not in the document]
Comment (pinpong) [reply lib/flutter_web3_provider.dart:11]: If the return value is null then not instance ist detected.

**Comment by jpiabrantes** [comment]
> Hi,
> 
> check: [flutter/flutter#43809](https://github.com/flutter/flutter/issues/43809)

Sorry if this is not the right place to ask this.

I have used that link to allow extensions on the Chrome browser that Flutter launches. However, everytime it launches a new browser and I need to reinstall metamask every time. Can it launch a browser with metamask already in it?

**Comment by pinpong** [comment]
@jpiabrantes ```flutter run -d web-server``` is what you are looking for ;D

**Pull request merged**